sdk/identity/azure-identity/inc/azure/identity/managed_identity_credential.hpp

```diff
@@ -42,6 +47,34 @@ namespace Azure { namespace Identity {
     std::string ToString() const { return m_resourceId; }
   };
 
+  /**
+   * @brief Options for managed identity credential.
+   *
+   */
+  struct ManagedIdentityCredentialOptions final : public Core::Credentials::TokenCredentialOptions
+  {
+    /**
+     * @brief The object ID of the service principal used to authenticate a user-assigned managed
+     * identity. This is an alternative to providing a client ID, and isn't required for
+     * system-assigned managed identities.
+     */
+    std::string ObjectId;
+
+    /**
+     * @brief The resource ID used to authenticate a resource with a user-assigned managed identity.
+     * This is an alternative to providing a client ID, and isn't required for system-assigned
+     * managed identities.
+     */
+    ResourceIdentifier ResourceId;
```

Comment:
To be able to add this to the options, we need a default ctor on `ResourceIdentifier`. That is less than ideal (since it doesn't make sense to provide a ResourceIdentifier with an empty string). I am open to alternatives to solve this concern.

Reply:
You could wrap it in an `Optional`.

Reply:
That's a pretty good alternative. Will consider that. Thanks!

Reply:
Another option is to add constructors for the ManagedIdentityCredentialOptions which take a ResourceId and ObjectId, and NOT have a default constructor for ManagedIdentityCredentialOptions - I don't know if you need to have a default constructor for `ManagedIdentityCredentialOptions` (which is what drives the need for the default constructor).